src/compiler/transformers/esnext.ts

```diff
@@ -906,7 +906,14 @@ namespace ts {
                 return i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i;
                 function verb(n) { if (g[n]) i[n] = function (v) { return new Promise(function (a, b) { q.push([n, v, a, b]) > 1 || resume(n, v); }); }; }
                 function resume(n, v) { try { step(g[n](v)); } catch (e) { settle(q[0][3], e); } }
-                function step(r) { r.value instanceof __await ? Promise.resolve(r.value.v).then(fulfill, reject) : settle(q[0][2], r);  }
+                function step(r) {

```

Comment:
Per the [spec](https://tc39.github.io/ecma262/#sec-asyncgeneratoryield), `yield` will always await. We already handle this behavior for `yield*` via `__asyncDelegator`. Rather than changing the helper here, we could handle this by changing the emit for the async generator body to replace `yield x` with `yield yield __await(x)`.

Reply:
@rbuckton makes sense! I gave it a try and update the PR. Could you take another look?

Reply:
`yield yield __await` looks to me like you're rewriting `yield x` into `yield await x`. I'm not sure those are the right semantics, even if they appear to work.

Reply:
@Kovensky, they actually are the correct semantics [per the spec](https://tc39.github.io/ecma262/#sec-asyncgeneratoryield), specifically in **25.5.3.7 AsyncGeneratorYield**:

> 5. Set _value_ to ? Await(_value_).